### PR TITLE
[fix](index compaction)skip index compaction when no output segment

### LINKED ISSUE #24466: [Bug] index compaction core
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

be.out
```
*** Query id: 0-0 ***
*** Aborted at 1694775962 (unix time) try "date -d @1694775962" if you are using GNU date ***
*** Current BE git commitID: d24f3efd4a ***
*** SIGSEGV address not mapped to object (@0x8) received by PID 1060572 (TID 1073508 OR 0x7f70c78ff700) from PID 8; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/common/signal_handler.h:413
 1# 0x00007F7510C08B50 in /lib64/libc.so.6
 2# std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/basic_string.h:6148
 3# doris::Compaction::do_compaction_impl(long) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:443
 4# doris::Compaction::do_compaction(long) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:124
 5# doris::FullCompaction::execute_compact_impl() at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/full_compaction.cpp:81
 6# doris::Compaction::compact() at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:100
 7# doris::CompactionAction::_execute_compaction_callback(std::shared_ptr<doris::Tablet>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/http/action/compaction_action.cpp:270
 8# std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status> >::_M_invoke(std::_Any_data const&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
 9# std::__future_base::_State_baseV2::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/future:575
10# __pthread_once_slow in /lib64/libpthread.so.0
11# std::__future_base::_State_baseV2::_M_set_result(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>, bool) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/future:411
12# std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run() at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/future:1447
13# execute_native_thread_routine at ../../../../../libstdc++-v3/src/c++11/thread.cc:84
14# start_thread in /lib64/libpthread.so.0
15# __clone in /lib64/libc.so.6
```

debug version's be.out
```
F20230915 18:51:19.122366 920664 compaction.cpp:440] Check failed: dest_index_files.size() > 0
*** Check failure stack trace: ***
    @     0x5590e5123416  google::LogMessage::SendToLog()
    @     0x5590e511f9e0  google::LogMessage::Flush()
    @     0x5590e5123c59  google::LogMessageFatal::~LogMessageFatal()
    @     0x5590d8834cd4  doris::Compaction::do_compaction_impl()
    @     0x5590d8832cb4  doris::Compaction::do_compaction()
    @     0x5590d8ccdab2  doris::FullCompaction::execute_compact_impl()
    @     0x5590d88328c0  doris::Compaction::execute_compact()
    @     0x5590d88327e3  doris::Compaction::compact()
    @     0x5590d98be9b3  doris::CompactionAction::_execute_compaction_callback()
    @     0x5590d98c0834  doris::CompactionAction::_handle_run_compaction()::$_1::operator()()
    @     0x5590d98c07d0  std::__invoke_impl<>()
    @     0x5590d98c07a0  _ZSt10__invoke_rIN5doris6StatusERZNS0_16CompactionAction22_handle_run_compactionEPNS0_11HttpRequestEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3$_1JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EESF_E4
typeEOSG_DpOSH_
    @     0x5590d98c0767  _ZZNSt13__future_base11_Task_stateIZN5doris16CompactionAction22_handle_run_compactionEPNS1_11HttpRequestEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3$_1SaIiEFNS1_6StatusEvEE6_M_runEvENKUlvE_clEv
    @     0x5590d98c0643  _ZNKSt13__future_base12_Task_setterISt10unique_ptrINS_7_ResultIN5doris6StatusEEENS_12_Result_base8_DeleterEEZNS_11_Task_stateIZNS3_16CompactionAction22_handle_run_compactionEPNS3_11HttpRequestEPNSt7__cxx1112basic
_stringIcSt11char_traitsIcESaIcEEEE3$_1SaIiEFS4_vEE6_M_runEvEUlvE_S4_EclEv
    @     0x5590d98c05f0  _ZSt13__invoke_implISt10unique_ptrINSt13__future_base7_ResultIN5doris6StatusEEENS1_12_Result_base8_DeleterEERNS1_12_Task_setterIS8_ZNS1_11_Task_stateIZNS3_16CompactionAction22_handle_run_compactionEPNS3_11HttpReq
uestEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3$_1SaIiEFS4_vEE6_M_runEvEUlvE_S4_EEJEET_St14__invoke_otherOT0_DpOT1_
    @     0x5590d98c0578  _ZSt10__invoke_rISt10unique_ptrINSt13__future_base12_Result_baseENS2_8_DeleterEERNS1_12_Task_setterIS0_INS1_7_ResultIN5doris6StatusEEES3_EZNS1_11_Task_stateIZNS7_16CompactionAction22_handle_run_compactionEPNS7_11
HttpRequestEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3$_1SaIiEFS8_vEE6_M_runEvEUlvE_S8_EEJEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EESU_E4typeEOSV_DpOSW_
    @     0x5590d98c0470  _ZNSt17_Function_handlerIFSt10unique_ptrINSt13__future_base12_Result_baseENS2_8_DeleterEEvENS1_12_Task_setterIS0_INS1_7_ResultIN5doris6StatusEEES3_EZNS1_11_Task_stateIZNS8_16CompactionAction22_handle_run_compacti
onEPNS8_11HttpRequestEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEE3$_1SaIiEFS9_vEE6_M_runEvEUlvE_S9_EEE9_M_invokeERKSt9_Any_data
    @     0x5590d901c2f8  std::function<>::operator()()
    @     0x5590d901c0b9  std::__future_base::_State_baseV2::_M_do_set()
    @     0x5590d901c29f  std::__invoke_impl<>()
    @     0x5590d901c20d  std::__invoke<>()
    @     0x5590d901c1d4  _ZZSt9call_onceIMNSt13__future_base13_State_baseV2EFvPSt8functionIFSt10unique_ptrINS0_12_Result_baseENS4_8_DeleterEEvEEPbEJPS1_S9_SA_EEvRSt9once_flagOT_DpOT0_ENKUlvE_clEv
    @     0x5590d901c1a7  _ZZNSt9once_flag18_Prepare_executionC1IZSt9call_onceIMNSt13__future_base13_State_baseV2EFvPSt8functionIFSt10unique_ptrINS3_12_Result_baseENS7_8_DeleterEEvEEPbEJPS4_SC_SD_EEvRS_OT_DpOT0_EUlvE_EERSI_ENKUlvE_clEv
    @     0x5590d901c171  _ZZNSt9once_flag18_Prepare_executionC1IZSt9call_onceIMNSt13__future_base13_State_baseV2EFvPSt8functionIFSt10unique_ptrINS3_12_Result_baseENS7_8_DeleterEEvEEPbEJPS4_SC_SD_EEvRS_OT_DpOT0_EUlvE_EERSI_ENUlvE_8__invok
eEv
    @     0x7f1df9adce67  __pthread_once_slow
    @     0x5590d9014dcb  __gthread_once()
    @     0x5590d901c039  std::call_once<>()
    @     0x5590d901bdd1  std::__future_base::_State_baseV2::_M_set_result()
    @     0x5590d98c0111  std::__future_base::_Task_state<>::_M_run()
    @     0x5590d98c48cb  std::packaged_task<>::operator()()
    @     0x5590d98c4895  std::__invoke_impl<>()
    @     0x5590d98c4855  std::__invoke<>()
*** Query id: 0-0 ***
*** Aborted at 1694775079 (unix time) try "date -d @1694775079" if you are using GNU date ***
*** Current BE git commitID: e9b5e037fe ***
*** SIGABRT unknown detail explain (@0x421000dda78) received by PID 907896 (TID 920664 OR 0x7f199b5ff700) from PID 907896; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/common/signal_handler.h:413
 1# 0x00007F1DFA4D9B50 in /lib64/libc.so.6
 2# __GI_raise in /lib64/libc.so.6
 3# __GI_abort in /lib64/libc.so.6
 4# 0x00005590E512DC2D in /mnt/disk2/luen/mydoris/be/lib/doris_be
 5# 0x00005590E511FF3A in /mnt/disk2/luen/mydoris/be/lib/doris_be
 6# google::LogMessage::SendToLog() in /mnt/disk2/luen/mydoris/be/lib/doris_be
 7# google::LogMessage::Flush() in /mnt/disk2/luen/mydoris/be/lib/doris_be
 8# google::LogMessageFatal::~LogMessageFatal() in /mnt/disk2/luen/mydoris/be/lib/doris_be
 9# doris::Compaction::do_compaction_impl(long) in /mnt/disk2/luen/mydoris/be/lib/doris_be
10# doris::Compaction::do_compaction(long) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:123
11# doris::FullCompaction::execute_compact_impl() at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/full_compaction.cpp:81
12# doris::Compaction::execute_compact() at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:105
13# doris::Compaction::compact() at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/olap/compaction.cpp:100
14# doris::CompactionAction::_execute_compaction_callback(std::shared_ptr<doris::Tablet>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at /mnt/disk2/luen/develop/workspace/incubator-doris/be/src/
http/action/compaction_action.cpp:270
15# doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1::operator()() const at /mnt/disk2/luen/develop/workspace/incubator-doris/be/sr
c/http/action/compaction_action.cpp:137
16# doris::Status std::__invoke_impl<doris::Status, doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1&>(std::__invoke_other, doris::
CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../..
/../../include/c++/11/bits/invoke.h:61
17# std::enable_if<is_invocable_r_v<doris::Status, doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1&>, doris::Status>::type std::__
invoke_r<doris::Status, doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1&>(doris::CompactionAction::_handle_run_compaction(doris::H
ttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:114
18# std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_ru
n()::{lambda()#1}::operator()() const at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/future:1444
19# std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpReque
st*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>::operator()() const at /mnt/disk2/luen/develop/tools/ldb_toolchain
/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/future:1362
20# std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter> std::__invoke_impl<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::_
_future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::
__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>&>(std::__invoke_other, std::__future_base::_Task_setter<std::unique_ptr<std:
:__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<ch
ar>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.
h:61
21# std::enable_if<is_invocable_r_v<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_
base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int
>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>&>, std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> >::type std::__invoke_r<std::unique_ptr<std::__future_base::_Result_base, std::_
_future_base::_Result_base::_Deleter>, std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_han
dle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>&>(std::__future_base::_Task_set
ter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<c
har, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, doris::Status ()>::_M_run()::{lambda()#1}, doris::Status>&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../inc
lude/c++/11/bits/invoke.h:114
22# std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<doris::Status>, std::__future_base::
_Result_base::_Deleter>, std::__future_base::_Task_state<doris::CompactionAction::_handle_run_compaction(doris::HttpRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)::$_1, std::allocator<int>, dor
is::Status ()>::_M_run()::{lambda()#1}, doris::Status> >::_M_invoke(std::_Any_data const&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
23# std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>::operator()() const at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/
c++/11/bits/std_function.h:560
24# std::__future_base::_State_baseV2::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64
-linux-gnu/11/../../../../include/c++/11/future:571
25# void std::__invoke_impl<void, void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, st
d::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::__invoke_memfun_deref, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_b
ase::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&
) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:74
26# std::__invoke_result<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::functi
on<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>::type std::__invoke<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base,
std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(void (std::__future_bas
e::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Resu
lt_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&) at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96
27# std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std
::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::once_flag&, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__
future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)::{lambda()#1}::operator(
)() const at /mnt/disk2/luen/develop/tools/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/mutex:778
28# std::once_flag::_Prepare_execution::_Prepare_execution<std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*)
```

### What You Expected?

success.

### How to Reproduce?

1. create table
```sql
CREATE TABLE IF NOT EXISTS mow
(
    `user_id` LARGEINT NOT NULL COMMENT "用户id",
    `username` VARCHAR(50) NOT NULL COMMENT "用户昵称",
    `city` VARCHAR(20) COMMENT "用户所在城市",
    `age` SMALLINT COMMENT "用户年龄",
    index index_city (city) using INVERTED PROPERTIES("parser" = "english")
)
UNIQUE KEY(`user_id`, `username`)
DISTRIBUTED BY HASH(`user_id`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"enable_unique_key_merge_on_write" = "true"
);
```
2. insert data
```sql
insert into mow values("1", "andy", "beijing", 10);
```
3. delete data
```sql
delete from mow where user_id=1;
```
4. force full compaction

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

Issue Number: close #24466

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -417,56 +417,65 @@ Status Compaction::do_compaction_impl(int64_t permits) {
...
-        }
-
-        // dest index files
-        // format: rowsetId_segmentId
-        std::vector<std::string> dest_index_files(dest_segment_num);
-        for (int i = 0; i < dest_segment_num; ++i) {
-            auto prefix = dest_rowset_id.to_string() + "_" + std::to_string(i);
-            dest_index_files[i] = prefix;
-        }
+        if (dest_segment_num > 0) {
```
Comment (eldenmoon) [inline be/src/olap/compaction.cpp:420]: could we add a test case?

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**